src/components/Channel/Channel.tsx

```diff
@@ -781,54 +785,113 @@ const ChannelInner = <
   const jumpToFirstUnreadMessage = useCallback(
     async (queryMessageLimit = 100) => {
       if (!(client.user && channelUnreadUiState?.unread_messages)) return;
-      if (!channelUnreadUiState?.last_read_message_id) {
-        addNotification(t('Failed to jump to the first unread message'), 'error');
-        return;
+      let lastReadMessageId = channelUnreadUiState?.last_read_message_id;
+      let firstUnreadMessageId = channelUnreadUiState?.first_unread_message_id;
+      let isInCurrentMessageSet = false;
+      let hasMoreMessages = true;
+
+      if (firstUnreadMessageId) {
+        const result = findInMsgSetById(firstUnreadMessageId, channel.state.messages);
+        isInCurrentMessageSet = result.index !== -1;
+      } else if (lastReadMessageId) {
+        const result = findInMsgSetById(lastReadMessageId, channel.state.messages);
+        isInCurrentMessageSet = !!result.target;
+        firstUnreadMessageId =
+          result.index > -1 ? channel.state.messages[result.index + 1]?.id : undefined;
+      } else {
+        const lastReadTimestamp = channelUnreadUiState.last_read.getTime();
+        const { index: lastReadMessageIndex, target: lastReadMessage } = findInMsgSetByDate(
+          channelUnreadUiState.last_read,
+          channel.state.messages,
+          true,
+        );
+
+        if (lastReadMessage) {
+          firstUnreadMessageId = channel.state.messages[lastReadMessageIndex + 1]?.id;
+          isInCurrentMessageSet = !!firstUnreadMessageId;
+          lastReadMessageId = lastReadMessage.id;
+        } else {
```

Comment:
In this block there's a dispatch for "setLoadingMore" but  no "loadMoreFinished" apart from the one in the `catch` block - is that expected? I'd add it to a `finally` block to have it run either way (success or error).

Reply:
cannot use finally as hasMoreMessages is updated later on. I have adjusted the code so that it calls `loadMoreFinished` for all the paths in the given condition branch.